app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt

```diff
@@ -64,6 +78,10 @@ class SettingsBasicsTest {
 
         // resetting modified features enabled setting to default
         featureSettingsHelper.resetAllFeatureFlags()
+
+        if (localeListIdlingResource != null) {
```

Comment:
Seems like this is used in the `switchLanguageTest` after which it will always be non null.
I understand there are cases where it won't be unregistered in the test and that's why we need to always do that here also.
A bit of a cleaner solution would be to use the idling resource in a try-finally block which would ensure we always cleanup while at the same time avoid duplicating the needed code or forgetting to duplicate it.
(A similar approach (`withInitialLocale(Locale, {})`) can be used for the locale setting method which needs to ensure it will set the original locale when the test finishes)
```kotlin
    /**
     * Run test with automatically registering idling resources and cleanup.
     *
     * @param idlingResources zero or more [IdlingResource] to be used when running [testBlock].
     * @param testBlock test code to execute.
     */
    private fun withIdlingResource(
        vararg idlingResources: IdlingResource,
        testBlock: () -> Unit
    ) {
        idlingResources.forEach {
            IdlingRegistry.getInstance().register(it)
        }

        try {
            testBlock()
        } finally {
            idlingResources.forEach {
                IdlingRegistry.getInstance().unregister(it)
            }
        }
    }
```

Reply:
Thanks! I'll create an issue for refactoring.